### PR TITLE
[fix]: [CDS-68205]: fetch SHA value of schemaVersion2 even when schemaVersion1 fails

### PR DESCRIPTION
**Pull request by Abhishek-CDC**
## Describe your changes
Following issues could be observed for docker artifact:
- Manifest API calls for schemaVersion1 and schemaVersion2 were in the same try catch block. It was possible for an image that manifest api for schemaVersion1 fails but successful for schemaVersion2. In such scenario SHA value for schemaVersion2 will be missed and there could be validation error because of digest mismatch.
- In case of tagRegex SHA and labels were not fetched.
- If labels were not present then SHA was also not consumed.

Made changes to call both APIs separately in different try catch block and handle these cases.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47201)

### CODE REVIEW

**Comment by Abhishek-CDC** [comment]
trigger smartchecks

File: 960-api-services/src/main/java/io/harness/artifacts/docker/service/DockerRegistryUtils.java [hunk 1]
```diff
@@ -58,23 +58,31 @@ public static String getDockerRegistryCredentials(ImageDetails imageDetails) {
...
-      if (artifactMetaInfo != null && shouldFetchDockerV2DigestSHA256) {
+      ArtifactMetaInfo artifactMetaInfoSchemaVersion1 = parseArtifactMetaInfoResponse(response, imageName);
+      if (artifactMetaInfoSchemaVersion1 != null) {
+        artifactMetaInfo = artifactMetaInfoSchemaVersion1;
+      }
+    } catch (Exception e) {
+      log.error("Unable to fetch artifact metainfo", e);
+    }
+
+    if (shouldFetchDockerV2DigestSHA256) {
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/artifacts/docker/service/DockerRegistryUtils.java:73]: Let's deprecate this usage if possible. Let's fetch both v1 and v2 like we do in other artifact sources.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 960-api-services/src/main/java/io/harness/artifacts/docker/service/DockerRegistryUtils.java (hunk 1 above)
Comment (Abhishek-CDC) [reply 960-api-services/src/main/java/io/harness/artifacts/docker/service/DockerRegistryUtils.java:73]: this method has multiple usages, with both true and false value. Not to break existing flow for others, made change to send always true for LastSuccessfulBuild

**Comment by Abhishek-CDC** [comment]
trigger utAll

**Comment by Abhishek-CDC** [comment]
trigger utAll

**Comment by Abhishek-CDC** [comment]
trigger uts

**Comment by Abhishek-CDC** [comment]
trigger uts

**Comment by Abhishek-CDC** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]

**Review by vivekkarnatiharness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, 2 of 3 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47201)* status: 5 file(s) to review. 1. unresolved discussions.

**Comment by Abhishek-CDC** [comment]
trigger review

**Pull request merged**